pull_data

```diff
@@ -0,0 +1,273 @@
+					popd > /dev/null
+				done
+
+			fi
+
+			if [[ $tests == *"speccpu2017"* ]]; then
+				pushd $top_dir > /dev/null
+				wget -r -np --spider -l1  -P . http://${pbench_results_locacation}/${1}/$sys/$tests 2> /dev/null
+				popd > /dev/null
+
+				for sdir in `ls -d results_speccpu*`; do
+					#
+					#  Pulls the index.html file.
+					#
+					mkdir -p $sdir/run_results
+					pushd $sdir/run_results
+					wget http://${pbench_results_locacation}/${1}/${sys}/${tests}/${sdir}/run_results/ 2> /dev/null
+					file_check=""
+					if [ -f "index.html" ]; then
+						file_check="index.html"
+					fi
+					if [ -f "${sdir}" ]; then
+						file_check=$sdir
+					fi
+					if [[ $file_check != "" ]]; then
+						file_to_pull=`grep csv $file_check | cut -d'<' -f 7 | cut -d'"' -f 2`
+						echo file_to_pull $file_to_pull
+						if [[ $file_to_pull != "" ]]; then
+							for get_file in $file_to_pull; do
+								wget http://${pbench_results_locacation}/${1}/${sys}/${tests}/$sdir/run_results/$get_file 2> /dev/null
+							done
+						fi
+						rm $file_check
+					fi
+					popd
+				done
+			fi
+			popd > /dev/null
+		done
+		popd > /dev/null
+	done
+}
+
+#
+# Now organize the data based on test then system class.  Also sort them (at least for aws).
+#
+organize_results()
+{
+	#
+	# List of tests we know about.
+	#
+	test_names="results_uperf.csv results_specjbb results_pig results_streams results_linpack results_hammerdb_maria results_hammerdb_pg  results_hammerdb_mssql fio_run results_hpl etcd speccpu"
+	#
+	# Get all the csv files present
+	#
+	find ${pbench_results_locacation} -print | grep csv > results_files
+
+	#
+	# Get the system prefixes. This is generic.
+	#	
+	systems=`cut -d'/' -f 4 results_files | sort -u`
+	series_list=""
+	separ=""
+	#
+	# If xlarge is in the system name, it is AWS.  We may
+	# have to deal with other AWS names later, but worry about
+	# it then.
+	#
+	if [[ $systems == *"xlarge"* ]]; then
+		#
+		# AWS systems
+		#
+		sys_type="aws"
+		#
+		# Build the series list (m5, m5a, i3en...)
+		#
+		type=`cut -d'/' -f 4 results_files | cut -d'.' -f 1 | sort -u`
+		for i in $type; do
+			grep ${i}"\." results_files > ${i}_series
+			series_list=${series_list}${separ}${i}_series
+			separ=" "
+		done
+	else
+		echo $systems is unknown.
+		exit 1
+	fi
+	rm results_locations 2> /dev/null
+	rm results_missing_tests 2> /dev/null
+	rm results_found_tests 2> /dev/null
+	#
+	# Walk through the tests and systems and build the appropriate results file
+	#
+	series_ordered=""
+	for tn in $test_names; do
+		echo "test ${tn}" >> results_locations
+		echo "test ${tn}" >> found_tests
+		for series in $series_list; do
+			if [[ $sys_type == "aws" ]]; then
+				#
+				# Obtain the system names, and sort it based on the xlarge value.
+				# 2xlarge, 4xlarge ....
+				#
+				series_ordered=`cut -d/ -f 4 $series | sort -u | sed "s/\./ /g" | sort -n -k2 | sed "s/ /./g"`
+			fi
+			if [[ $sys_type == "azure" ]]; then
+				echo Need to handle azure
```

Comment:
What kind of info do you need to add Azure support?  It'd be best to just roll this all in at once, but if it's too long a project then I guess it can be done after.